src/rest_api/classes/laboratory/widgets/gene_classes.clj

```diff
@@ -0,0 +1,59 @@
+(ns rest-api.classes.laboratory.widgets.gene-classes
+  (:require
+    [datomic.api :as d]
+    [pseudoace.utils :as pace-utils]
+    [rest-api.classes.generic-fields :as generic]))
+
+(defn former-gene-classes [laboratory]
+  (let [db (d/entity-db laboratory)
+        data (->> (d/q '[:find ?gene-class ?description
+                         :in $ ?laboratory
+                         :where 
+                         [?laboratoryent :gene-class.former-designating-laboratory/laboratory ?laboratory]
+                         [?gc :gene-class/former-designating-laboratory ?laboratoryent]
+                         [?gc :gene-class/id ?gene-class]
+                         [?gc :gene-class/description ?description]]
+                       db (:db/id laboratory))
+                  (map (fn [oid]
```

Comment:
Consider destructuring each result row rather than using `(get result index)`. 
```clojure
(map (fn [result]
            (let [[geme-class description] result]
               ....)
```

Reply:
Good to know, thanks, fixed.

Reply:
About the main point about the query results being mapped into the same kind of hash, one tag has a keyword :former_gene_class while the other has :gene_class .  Since they're slightly different, would you prefer a single function passing [result] + a variable for the keyword being one or the other, applying (keyword) to it, and generating the map in a single function ?  I couldn't tell whether that was more readable or preferred.

Reply:
Having two separate functions for the two distinct endpoints is fine; I was alluding to the inline function passed to `map` in the thread-macro in both functions.
So this:
```clojure
(fn [result]
  (let [[gene-class description] result]
    {:description description
     :gene_class
     {:taxonomy "all"
      :class "gene_class"
      :label gene-class
      :id gene-class}})
```
could be factored-out and given a name e.g `(defn- process-result [result] ...)` or something like that.
Used then in both `former-gene-class` and `gene-class` end-point fns : `(map process-result)`

Reply:
Ah, yes, that's what I meant.  In gene-class the map is like : 
{:description description
     :gene_class
     {:taxonomy "all"
but in former-gene-class, the :gene_class tag becomes :former_gene_class  
{:description description
     :former_gene_class
     {:taxonomy "all"
so it would either need two separate functions, or pass in an extra variable to change into the (keyword), and I couldn't tell whether that was better / easier to understand.